components/spell-check.R

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env Rscript
+#
+# Run spell check and save results
+
+library(magrittr)
+
+# Read in dictionary
+dictionary <- readLines(file.path('components', 'dictionary.txt'))
+
+# Only declare Rmd files
+files <- list.files(pattern = 'Rmd$', recursive = TRUE, full.names = TRUE)
+
+# Run spell check
+sp_errors <- spelling::spell_check_files(files, ignore = dictionary) %>% 
+  data.frame() %>%
+  tidyr::unnest(cols = found)
+
+# Print out how many spell check errors
+write(nrow(sp_errors), stdout())
+
+# Save spell errors to file temporarily
+readr::write_tsv(sp_errors, file.path('components', 'spell_check_results.tsv'))
```

Comment:
I would suggest either making this an option or outputting directly to the current directory. No guarantees of where this will be run from, so expecting a folder can seem dangerous.